src/components/Evolution.ts

```diff
@@ -3,11 +3,61 @@ import { Component } from "eightbittr/lib/Component";
 import { FullScreenPokemon } from "../FullScreenPokemon";
 import { IPokemon } from "./Battles";
 import { 
-    IPokemonEvolution, IPokemonEvolutionByLevel, IPokemonEvolutionByStats, 
-    IPokemonEvolutionRequirements
+    IPokemonEvolution, IPokemonEvolutionByItem, IPokemonEvolutionByLevel, 
+    IPokemonEvolutionByStats, IPokemonEvolutionByTrade, IPokemonEvolutionRequirement
 } from "./constants/Pokemon";
 
 /**
+ * Holds arguments used in IRequirementHander functions.
+ */
+export interface IHandlerArgs<T1, T2> {
```

Comment:
I'm not a fan of the name `IHandlerArgs`. It's pretty generic. `IRequirementHandlerArgs` at least mentions that it's for requirements. Maybe something like that or `IEvolutionRequirementArgs`?